src/core/src/nsm_client.cpp

```diff
@@ -20,101 +20,414 @@
+				const QString sDrumkitNameXML = H2Core::LocalFileMng::readXmlString( drumkitInfoNode, "name", "" );
+	
+				if ( sDrumkitNameXML == sDrumkitName ) {
+					bRelinkDrumkit = false;
+				} else {
+					NsmClient::printError( QString( "Linked [%1] and loaded [%2] drumkit do not match." )
+										   .arg( sDrumkitNameXML )
+										   .arg( sDrumkitName ) );
+				}
+			}
+		} else {
+			NsmClient::printError( "Symlink does not point to valid drumkit." );
+		}				   
+	}
+	
+	// The symbolic link either does not exist, is not valid, or does
+	// point to the wrong location. Remove it and create a fresh one.
+	if ( bRelinkDrumkit ){
+		NsmClient::printMessage( "Relinking drumkit" );
+		QFile linkedDrumkitFile( sLinkedDrumkitPath );
+		
+		if ( linkedDrumkitFile.exists() ) {
+			if ( linkedDrumkitPathInfo.isDir() &&
+				 ! linkedDrumkitPathInfo.isSymLink() ) {
+				// Move the folder so we don't use the precious old
+				// drumkit. But in order to use it again, it has to be
+				// renamed to 'drumkit' manually again.
+				QDir oldDrumkitFolder( sLinkedDrumkitPath );
+				if ( ! oldDrumkitFolder.rename( sLinkedDrumkitPath,
+												QString( "%1/drumkit_old" ).arg( name ) ) ) {
+					NsmClient::printError( QString( "Unable to rename drumkit folder [%1]." )
+										   .arg( sLinkedDrumkitPath ) );
+					return;
+				}
+			} else {
+				if ( !linkedDrumkitFile.remove() ) {
+					NsmClient::printError( QString( "Unable to remove symlink to drumkit [%1]." )
+										   .arg( sLinkedDrumkitPath ) );
+					return;
+				}
+			}
+		}
+		
+		// Figure out the actual path to the drumkit. We will search
+		// the user drumkits first and the system ones second.
+		QString sDrumkitAbsPath( "" );
+		const QStringList drumkitListUsr = H2Core::Filesystem::usr_drumkit_list();
+		for ( auto ssName : drumkitListUsr ) {
+			if ( ssName == sDrumkitName ) {
+				sDrumkitAbsPath = H2Core::Filesystem::usr_drumkits_dir() + ssName;
+			}
+		}
+		
+		if ( sDrumkitAbsPath.isEmpty() ) {
+			const QStringList drumkitListSys = H2Core::Filesystem::sys_drumkit_list();
+			for ( auto ssName : drumkitListSys ) {
+				if ( ssName == sDrumkitName ) {
+					sDrumkitAbsPath = H2Core::Filesystem::sys_drumkits_dir() + ssName;
+				}
+			}
+		}
+		
+		if ( sDrumkitAbsPath.isEmpty() ) {
+			// Something went wrong. We skip the linking.
+			NsmClient::printError( QString( "No drumkit named [%1] could be found." )
+								   .arg( sDrumkitName ) );
+		} else {
+			
+			// Actual linking.
+			QFile targetPath( sDrumkitAbsPath );
+			if ( !targetPath.link( sLinkedDrumkitPath ) ) {
+				NsmClient::printError( QString( "Unable to link drumkit [%1] to [%2]." )
+									   .arg( sLinkedDrumkitPath )
+									   .arg( sDrumkitAbsPath ) );
+			}
+		}
+	}
 }
 
-void* nsm_processEvent(void* data)
-{
-	nsm_client_t* nsm = (nsm_client_t*) data;
+void NsmClient::printError( const QString& msg ) {
+	std::cerr << "[\033[30mHydrogen\033[0m]\033[31m "
```

Comment:
Errorlog?

Reply:
As pointed out above the log of the logger won't be visible when called by the NSM daemon since it's not directed to stderr. 

This is also the reason I introduced all this raw Qt-based file copying instead of using the corresponding functions from `Filesystem`. Problems wouldn't show up in the log of non-session-manager